File: recettes.pl
% Faits : Définition des recettes et de leurs ingrédients
recette(pizza, farine).
recette(pizza, eau).
recette(pizza, sel).
recette(pizza, levure).
recette(pizza, tomate).
recette(pizza, fromage).

recette(salade, laitue).
recette(salade, tomate).
recette(salade, concombre).
recette(salade, vinaigre).
recette(salade, huile).

recette(pates_carbonara, pates).
recette(pates_carbonara, creme).
recette(pates_carbonara, lardons).
recette(pates_carbonara, fromage).
recette(pates_carbonara, sel).
recette(pates_carbonara, poivre).

recette(omelette, oeufs).
recette(omelette, sel).
recette(omelette, poivre).
recette(omelette, fromage).
recette(omelette, herbes).

recette(sandwich_jambon_beurre, pain).
recette(sandwich_jambon_beurre, beurre).
recette(sandwich_jambon_beurre, jambon).
recette(sandwich_jambon_beurre, salade).

% Vérifie
recettedisponible(Recette, Ingredients) :-
    recette(Recette, ),                   % Vérifie que la recette existe
    forall(recette(Recette, Ing), member(Ing, Ingredients)). % Vérifie tous les ingrédients

File: recettes.py
recettes = {
    "Pizza": {"farine", "eau", "sel", "levure", "tomate", "fromage"},
    "Salade": {"laitue", "tomate", "concombre", "vinaigre", "huile"},
    "Pâtes Carbonara": {"pâtes", "crème", "lardons", "fromage", "sel", "poivre"},
    "Omelette": {"œufs", "sel", "poivre", "fromage", "herbes"},
    "Sandwich Jambon-Beurre": {"pain", "beurre", "jambon", "salade"}
}

def recettes_disponibles(ingredients_disponibles):
    recettes_possibles = []
    for recette, ingredients_recette in recettes.items():
        if ingredients_recette.issubset(ingredients_disponibles):
            recettes_possibles.append(recette)
    return recettes_possibles

test_ingredients = [
    {"farine", "eau", "sel", "levure", "tomate", "fromage"},  # Pizza
    {"pain", "beurre", "jambon"},  # Ne peut pas faire de sandwich car il manque la salade
    {"œufs", "sel", "poivre", "fromage", "herbes"},  # Omelette
    {"laitue", "tomate", "concombre", "vinaigre", "huile"}  # Salade
]

for i, ingredients in enumerate(test_ingredients, 1):
    print(f"Test {i} - Ingrédients disponibles: {ingredients}")
    recettes_possibles = recettes_disponibles(ingredients)
    if recettes_possibles:
        print("Recettes possibles :", ", ".join(recettes_possibles))
    else:
        print("Aucune recette réalisable")
